### ISSUE TITLE
Critical: PAT Fine-Tuning Uses Wrong Pretrained Weights (Data Leakage)

### ISSUE THREAD

**Issue by The-Obstacle-Is-The-Way** [Clarity-Digital-Twin/big-mood-detector#85]
We've discovered why we can't improve PAT depression detection beyond 0.593 AUC (Issue #60). We're using PAT models pretrained on 29k participants which **includes NHANES 2013-2014**, then fine-tuning on the same NHANES 2013-2014 dataset. This creates severe data leakage.

## The Problem

**The PAT authors confirmed**:
> "For NHANES data, we actually used the PAT models trained with (21k participants) instead of (29k participants). This dataset excludes participants from 2013-2014 NHANES so that there wouldn't be data leakage."

## Impact

- Model has memorized test data during pretraining
- Cannot learn new patterns or improve
- Current 0.593 AUC is artificially inflated
- Blocks Issue #60 resolution

## Evidence in Code

```python
# src/big_mood_detector/infrastructure/ml_models/pat_model.py:137
weights_path = Path(env_dir) / f"PAT-{size_suffix}_29k_weights.h5"

# src/big_mood_detector/infrastructure/fine_tuning/population_trainer.py:292
base_model_path: str = "weights/PAT-S_29k_weights.h5",
```

## Solution

1. Switch to 21k pretrained weights (we already have them)
2. Retrain depression head from scratch
3. Expect initial performance drop (this is good\!)
4. Target realistic 0.60-0.61 AUC

## Additional Implementation Differences Found

1. **No Log Transform**: PAT authors don't use `Log(x+1)` transform
2. **Conv1D Parameters**: kernel_size=3, padding='same', activation='relu' (we use 9, 0, None)
3. **Early Stopping**: Patience of 250 epochs (we stop too early at epoch 2)
4. **Learning Rate**: Consider lower rates if peaking at epoch 2

**Related to**: #60 
**Priority**: CRITICAL - This affects all PAT-based predictions and blocks further improvements

**Comment by The-Obstacle-Is-The-Way** [comment]
✅ RESOLVED - Fixed data leakage and trained with correct pretrained weights\!

## What We Fixed:
1. **Switched to 21k Pretrained Weights**: No NHANES 2013-2014 contamination
2. **Removed Data Leakage**: Test set never seen during pretraining or training
3. **Implemented Paper's Architecture**: Conv1D with correct parameters
4. **Extended Training**: Proper early stopping (37 epochs vs 2)

## Results:
- **Validation AUC: 0.6708** (exceeds paper's 0.625 target\!)
- **Test AUC: 0.5840** (honest evaluation without leakage)
- Previous buggy version: 0.5929 (with leakage)

## Key Changes Made:
1. Used PAT-L_21k_weights.h5 (excludes NHANES 2013-2014)
2. Proper Conv1D: kernel_size=3, padding='same', activation='relu'
3. No log transform on activity counts
4. Cosine annealing LR schedule
5. Early stopping patience increased

## Evidence:
- Training script: scripts/pat_training/train_pat_conv_l_final.py
- Training log: scripts/pat_training/training_runs/2025-07-31_final_corrected/pat_conv_l_final.log
- Commit: 0e53e7d

Successfully achieved better-than-paper validation performance with honest evaluation\!

**Comment by The-Obstacle-Is-The-Way** [comment]
CORRECTION: The paper's 0.625 is TEST performance, not validation.

## Accurate Comparison:
- **Paper's TEST AUC**: 0.625 
- **Our TEST AUC**: 0.5840 (0.041 gap)
- **Our Validation AUC**: 0.6708 (paper doesn't report validation)

The 0.041 test performance gap is likely due to:
1. Different data splits (they use stratified sampling with replacement)
2. Preprocessing (Savitzky-Golay smoothing)
3. Standardization methodology
4. Linear probe hyperparameters

We successfully fixed the data leakage and achieved honest evaluation\!
